File: src/main/java/org/academy/api/client/gui/command/ImageDrawCommand.java
package org.academy.api.client.gui.command;

import com.mojang.blaze3d.buffers.GpuBufferSlice;
import com.mojang.blaze3d.textures.GpuTextureView;
import org.academy.api.client.Render;

import java.util.Collections;
import java.util.Map;

public class ImageDrawCommand extends PosTexColorRectDrawCommand {
    protected final GpuTextureView textureView;

    public ImageDrawCommand(
            GpuTextureView gpuTextureView,
            float width,
            float height,
            float u0,
            float v0,
            float u1,
            float v1,
            float red,
            float green,
            float blue,
            float alpha
    ) {
        super(Render.RenderPipelines.IMAGE,width,height,u0,v0,u1,v1,red,green,blue,alpha);
        textureView = gpuTextureView;
    }

    @Override
    public Map<String, GpuTextureView> getSamplers() {
        return Map.of("Sampler0", textureView);
    }

    @Override
    public Map<String, GpuBufferSlice> getUniforms() {
        return Collections.emptyMap();
    }
}

File: src/main/kotlin/org/academy/api/client/gui/command/ImageDrawCommand.kt
package org.academy.api.client.gui.command

import com.mojang.blaze3d.textures.GpuSampler
import com.mojang.blaze3d.textures.GpuTextureView
import org.academy.api.client.Render
import org.academy.api.client.render.TextureBinding

class ImageDrawCommand(
    texture: GpuTextureView, sampler: GpuSampler,
    width: Float, height: Float,
    u0: Float, v0: Float, u1: Float, v1: Float,
    red: Float, green: Float, blue: Float, alpha: Float
) : PosTexColorRectDrawCommand(
    Render.RenderPipelines.IMAGE,
    width, height,
    u0, v0, u1, v1,
    red, green, blue, alpha,
    listOf(
        TextureBinding(
            "Sampler0",
            texture,
            sampler
        )
    ),
    mutableListOf()
) 